pkg/controllers/scheduling/placeholder_controller.go

```diff
@@ -0,0 +1,231 @@
+package scheduling
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	cache "k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	clusterclient "github.com/open-cluster-management/api/client/cluster/clientset/versioned"
+	clusterinformerv1alpha1 "github.com/open-cluster-management/api/client/cluster/informers/externalversions/cluster/v1alpha1"
+	clusterlisterv1 "github.com/open-cluster-management/api/client/cluster/listers/cluster/v1"
+	clusterlisterv1alpha1 "github.com/open-cluster-management/api/client/cluster/listers/cluster/v1alpha1"
+	clusterapiv1alpha1 "github.com/open-cluster-management/api/cluster/v1alpha1"
+)
+
+// decisionPlaceholderController creates empty decisions as placeholders in status of PlacementDecisions.
```

Comment:
I'll see how the impl falls out, but here's my naive thoughts on how I would try to structure this
1. given an update to a placementdecision, queue the key for its placement
2. given an update to a placement, queue the key for it
3. given an update to a managedcluster, queue the key for all placements it is currently bound to (this will allow for quick removals).  I would do this in a later PR, it's not trivial to build.
4. given a update of a managedcluster, queue the keys for all placements that mention its labels, managedclusterset, or claims. I would do this in a later PR, it's not trivial to build.
5. given a placement, I would compute the sorted list of all managedclusters
6. given the sorted list, I would diff against the placementdecision and create, update, and delete accordingly
7. I would resync all placements every X minutes to compensate for not having 3 or 4 for a while.